bin/describe/describe_workspace.ml

```diff
@@ -593,15 +593,16 @@ let term : unit Term.t =
              Path.Source.(relative root) (Common.prefix_target common dir)
            in
            let s = Path.source p in
-           if not @@ Path.exists s then
-             User_error.raise
-               [ Pp.textf "No such file or directory: %s" (Path.to_string s) ];
-           if not @@ Path.is_directory s then
+           match Path.stat s with
+           | Ok { Unix.st_kind = Unix.S_DIR; _ } -> Memo.return p
+           | Ok _ ->
              User_error.raise
                [ Pp.textf "File exists, but is not a directory: %s"
                    (Path.to_string s)
-               ];
-           Memo.return p))
+               ]
+           | Error _ ->
+             User_error.raise
+               [ Pp.textf "No such file or directory: %s" (Path.to_string s) ]))
```

Comment:
I wanted to highlight this actually. I don't think `Path.stat` is very useful since the `Error` type is just `Unix_error.Detailed.t` so we cannot match on the underlying Unix error. Should we change the type or perhaps just match on `Path.stat_exn` directly with `exception` patterns?

Reply:
What is the first element in the tuple of `Unix_error.Detailed.t` if not the underlying error?

Reply:
Huh, I misread earlier then. I'll fix this.